nomos-core/chain-defs/src/mantle/tx.rs

```diff
@@ -160,3 +278,364 @@ impl GasCost for SignedMantleTx {
+                Some(OpProof::Ed25519Sig(sig1)),
+                Some(OpProof::Ed25519Sig(sig2)),
+            ],
+            dummy_zk_signature(),
+        );
+
+        assert!(matches!(
+            result,
+            Err(VerificationError::InvalidSignature { op_index: 1 })
+        ));
+    }
+
+    #[test]
+    fn test_signed_mantle_tx_deserialize_with_valid_proof() {
+        let signing_key = SigningKey::from_bytes(&[1; 32]);
+        let blob_op = create_test_blob_op(&signing_key);
+        let mantle_tx = create_test_mantle_tx(vec![Op::ChannelBlob(blob_op)]);
+
+        let tx_hash = mantle_tx.hash().as_signing_bytes();
+        let signature = signing_key.sign(&tx_hash);
+
+        let signed_tx = SignedMantleTx::new(
+            mantle_tx,
+            vec![Some(OpProof::Ed25519Sig(signature))],
+            dummy_zk_signature(),
+        )
+        .unwrap();
+
+        // Serialize and deserialize
+        let serialized = serde_json::to_string(&signed_tx).unwrap();
+        let deserialized: Result<SignedMantleTx, _> = serde_json::from_str(&serialized);
+
+        assert!(deserialized.is_ok());
+        assert_eq!(deserialized.unwrap(), signed_tx);
+    }
+
+    #[test]
+    fn test_signed_mantle_tx_deserialize_with_missing_proof() {
+        let signing_key = SigningKey::from_bytes(&[1; 32]);
+        let blob_op = create_test_blob_op(&signing_key);
+        let mantle_tx = create_test_mantle_tx(vec![Op::ChannelBlob(blob_op)]);
+
+        let helper = SignedMantleTx {
+            mantle_tx,
+            ops_proofs: vec![None],
+            ledger_tx_proof: dummy_zk_signature(),
+        };
+
+        let serialized = serde_json::to_string(&helper).unwrap();
+        let deserialized: Result<SignedMantleTx, _> = serde_json::from_str(&serialized);
+
+        assert!(deserialized.is_err());
+        let err_msg = deserialized.unwrap_err().to_string();
+        assert!(err_msg.contains("MissingProof") || err_msg.contains("ChannelBlob"));
+    }
+
+    #[test]
+    fn test_signed_mantle_tx_deserialize_with_invalid_signature() {
+        let signing_key = SigningKey::from_bytes(&[1; 32]);
+        let wrong_key = SigningKey::from_bytes(&[2; 32]);
+        let blob_op = create_test_blob_op(&signing_key);
+        let mantle_tx = create_test_mantle_tx(vec![Op::ChannelBlob(blob_op)]);
+
+        let tx_hash = mantle_tx.hash().as_signing_bytes();
+        let wrong_signature = wrong_key.sign(&tx_hash);
+
+        let helper = SignedMantleTx {
+            mantle_tx,
+            ops_proofs: vec![Some(OpProof::Ed25519Sig(wrong_signature))],
+            ledger_tx_proof: dummy_zk_signature(),
+        };
+
+        let serialized = serde_json::to_string(&helper).unwrap();
+        let deserialized: Result<SignedMantleTx, _> = serde_json::from_str(&serialized);
+
+        assert!(deserialized.is_err());
+        let err_msg = deserialized.unwrap_err().to_string();
+        assert!(err_msg.contains("Invalid signature"));
+    }
+}
```

Comment:
nice tests, perhaps one more to check mismatched ops and ops_proofs lengths (should always fail)